futures-core/src/task/__internal/atomic_waker.rs

```diff
@@ -268,15 +268,18 @@ impl AtomicWaker {
                     // Release the lock. If the state transitioned to include
                     // the `WAKING` bit, this means that at least one wake has
                     // been called concurrently.
-                    //
+                    // 
                     // Start by assuming that the state is `REGISTERING` as this
-                    // is what we jut set it to.
+                    // is what we just set it to. If this holds, we know that no
+                    // other writes were performed in the meantime, so there is
+                    // nothing to acquire, but release the update. In case of
+                    // concurrent wakers, we need to acquire their releases.
                     let res = self.state.compare_exchange(
-                        REGISTERING, WAITING, AcqRel, Acquire);
+                        REGISTERING, WAITING, Release, Acquire);
```

Comment:
Errors with `a failure ordering can't be stronger than a success ordering`

Reply:
Ah, yes, makes sense, will update.